opencog/nlp/chatbot/sentence-matching.scm

```diff
@@ -0,0 +1,32 @@
+; Returns one or more sentences that are structurally similar to the input one.
```

Comment:
> 2) this is the wrong directory for this code. That chatbot directory
> contains an IRC chatbot interface, and nothing more. Your code has nothing
> to do with IRC, and so doesn't belong.
>
>
Yes, this makes sense.

While the proximate cause for implementing this fuzzy matching
functionality is to experiment with a simple/crude form of R2L-based
question-answering, the functionality itself is very general...

Semi-relatedly,
I would add also that the version being pushed is not really great in terms
of its AI functionality -- it returns too many false matches in many
cases.  Man Hin is working on an improvement to the matching algorithm
(using Atom probability values to determine the costs used in the dynamic
programming hypergraph matching), but that is still half-baked and will be
pushed later...

-- Ben